src/query/ts/m3db/convert.go

```diff
@@ -166,14 +192,20 @@ func seriesBlocksFromBlockReplicas(blockReplicas []blockReplica, seriesIterator
 			Replicas:       block.replicas,
 		}, nil)
 
-		series.Blocks[i] = SeriesBlock{
-			start:          filterValuesStart,
-			end:            filterValuesEnd,
-			seriesIterator: valuesIter,
-		}
+		// NB(braskin): we should be careful when directly accessing the series iterators.
+		// Instead, we should access them through the SeriesBlock.
+		blockBuilder.add(
+			&models.Bounds{
+				Start:    filterValuesStart,
+				Duration: block.blockSize,
+				StepSize: stepSize,
+			},
+			iter,
+			i == replicaLength,
```

Comment:
move this out into a variable for better readability

Reply:
Done